File: tools/base2To10_justCode.js
function base2to10 (str) {
    var n = 0
    var total = 0

    for (var i = str.length - 1; i >= 0; i = i - 1) {
        total = total + Number(str.charAt(i)) * 2**n
        n = n + 1
    }
    return total
}
console.log(base2to10("011001"))
console.log(base2to10("000000"))

File: tools/base2To10_justCode.py
def base2to10(str):
	n = 0
	total = 0

	for i in range(len(str) -1, -1, -1):
		total = total + int(str[i]) * 2**n
		n = n + 1
	return total

print(base2to10("011001"))